android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -626,31 +653,94 @@ public static boolean isPublic(UrlDevice urlDevice) {
   }
 
   /**
+   * Checks if device is PWS resolvable.
+   * @param urlDevice The device that is getting checked.
+   * @return If the device is resolvable or not.
+   */
+  public static boolean isResolvableDevice(UrlDevice urlDevice) {
+    String type;
+    try {
+      type = urlDevice.getExtraString(TYPE_KEY);
```

Comment:
You can use optExtraString(TYPE_KEY, "") and then do you don't have to wrap it in a try block